prov/rxm/src/rxm_cq.c

```diff
@@ -1195,8 +1195,15 @@ static void rxm_cq_read_write_error(struct rxm_ep *rxm_ep)
 		rx_buf = (struct rxm_rx_buf *)err_entry.op_context;
 		util_cq = rx_buf->ep->util_ep.rx_cq;
 		util_cntr = rx_buf->ep->util_ep.rx_cntr;
-		err_entry.op_context = rx_buf->recv_entry->context;
-		err_entry.flags = rx_buf->recv_entry->comp_flags;
+		if (rx_buf->recv_entry) {
+			err_entry.op_context = rx_buf->recv_entry->context;
+			err_entry.flags = rx_buf->recv_entry->comp_flags;
+		} else {
+			/* This branch may be reached if we are aborting the eagerly submitted
+			 * msg recv buffers for incoming tagged messages. In that case, there's
+			 * no context for the error. */
+			err_entry.op_context = NULL;
+		}
```

Comment:
We should return without writing any error to CQ in the else case. This internal error should not be propagated to app. May be move setting of util_cq and util_cntr into the True branch.

Reply:
The comment conflicts somewhat with the states being checked by the switch-case statements.  I.e. two of the case statements are related to rendezvous, but the problem path is related to eager messages.  Assuming the else clause is restricted to the case RXM_RX, maybe we should restructure the code so that the RXM_RX case handles the problem child and falls through otherwise.